edb/server/connpool/pool.py

```diff
@@ -226,11 +228,7 @@ def try_steal(
 
         return self.conn_stack.popleft()
 
-    async def acquire(self) -> C:
-        # There can be a race between a waiter scheduled for to wake up
-        # and a connection being stolen (due to quota being enforced,
-        # for example).  In which case the waiter might get finally
-        # woken up with an empty queue -- hence we use a `while` loop here.
+    async def try_acquire(self) -> typing.Optional[C]:
         self.conn_waiters_num += 1
         try:
             attempts = 0
```

Comment:
`attempts` should go with the while loop into the new outer `acquire()`, and pass in as a parameter.